src/Microsoft.SqlTools.ServiceLayer/ShowPlan/ShowPlanGraph/Node.cs

```diff
@@ -55,6 +57,186 @@ public int GroupIndex
+                    if (secondLine != null)
+                    {
+                        if (secondLine != firstLine)
+                        {
+                            // Enclose logical name in parenthesis.
+                            secondLine = SR.Parenthesis(secondLine);
+                        }
+                        else
+                        {
+                            // Don't show the second line if its value is the same as on the first line.
+                            secondLine = null;
+                        }
+                    }
+                }
+
+                return secondLine == null || secondLine.Length == 0
+                    ? firstLine
+                    : String.Format(CultureInfo.CurrentCulture, "{0}\n{1}", firstLine, secondLine);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets node name (visible through AA)
+        /// </summary>
+        public string Name2
+        {
+            get
+            {
+                // Make AA names compatible with the previos implementation
+                string name = this.DisplayName.Replace("\n", "/");
+                name = name.Replace("(", "");
+                name = name.Replace(")", "");
+                if (name.Length == 0)
+                {
+                    return SR.NodeDisplayPropertiesName1(this.ID);
+                }
+                else
+                {
+                    return SR.NodeDisplayPropertiesName2(name, this.ID);
+                }
+            }
+
+            set
+            {
+                ;
+            }
+        }
+
+        /// <summary>
+        /// Gets Node description
+        /// </summary>
+        [DisplayOrder(2), DisplayNameDescription(SR.Keys.OperationDescriptionShort, SR.Keys.OperationDescription)]
+        public string Description
+        {
+            get { return this.Operation.Description; }
+        }
+
+        /// <summary>
+        /// Gets the value that indicates Node parallelism.
+        /// </summary>
+        public bool IsParallel
+        {
+            get
+            {
+                object value = this["Parallel"];
+                return value != null ? (bool)value : false;
+            }
+        }
+
+        /// <summary>
+        /// Gets the value that indicates whether the Node has warnings.
+        /// </summary>
+        [Browsable(false)]
+        public bool HasWarnings
+        {
+            get
+            {
+                return this["Warnings"] != null;
+            }
+        }
+
+
+        /// <summary>
+        /// Gets the value that indicates whether the Node has critical warnings.
+        /// </summary>
+        private bool HasCriticalWarnings
+        {
+            get
+            {
+                if (this["Warnings"] != null)
+                {
+                    ExpandableObjectWrapper wrapper = this["Warnings"] as ExpandableObjectWrapper;
+                    if (wrapper["NoJoinPredicate"] != null)
```

Comment:
Why this specific property?

Reply:
Got this piece of code from SSMS. I haven't really investigated the logic of this property